app/javascript/features/lobby/display/BattleCardPlayer.jsx

```diff
@@ -92,14 +110,17 @@ export const BattleCardPlayer = ({ battleId, playerId, playerName, companyDoctri
   if (playerId && isCurrentPlayer) {
     // Filled spot by logged in player
     let readyContent
-    if ((battleState === FULL && ready) || battleState === GENERATING) {
+    if (battleState === GENERATING) {
       readyContent = <CheckIcon className={classes.clickableIcon} color="success" />
+    } else if (battleState === FULL && ready) {
+      readyContent = <CheckIcon className={classes.clickableIcon} onClick={handleUnreadyClick} color="success" />
     } else if (battleState === FULL && !ready) {
       readyContent = <Button variant="contained" type="submit" color="secondary" size="small"
                              className={classes.readyBtn} onClick={handleReadyClick} disabled={isPending}>Ready</Button>
-    } else if (battleState === INGAME) {
-      // TODO Abandon
-
+    } else if (battleState === INGAME && !abandoned) {
+      readyContent = <HighlightOffIcon className={classes.clickableIcon} onClick={handleAbandonClick} color="error"/>
+    } else if (battleState === INGAME && abandoned) {
+      readyContent = <CancelIcon  className={classes.clickableIcon} onClick={handleUnabandonClick} color="success"/>
```

Comment:
Please take a look at https://github.com/mma127/omg_rails/pull/85/files#diff-597f0e43adba681da3329a4c618913e9b0114327a5d2514df24cb3a1b6218332R97, we should use all abandonable states